cli/injector/cpu_pressure.go

```diff
@@ -18,10 +21,40 @@ var cpuPressureCmd = &cobra.Command{
 	PreRun: func(cmd *cobra.Command, args []string) {
 		// prepare spec
 		spec := v1beta1.CPUPressureSpec{}
+		podCores, err := getPodCores()
+		if err != nil {
+			log.Fatalw("error creating the node injector", "error", err)
+			return
+		}
+
+		cores, err := cpuset.Parse(podCores)
+		if err != nil {
+			log.Fatalw("error creating the node injector", "error", err)
+			return
+		}
 
+		log.Infow(fmt.Sprintf("target identified to be running on %d cores", cores.Size()), "cores", cores.ToSlice())
 		// create injector
 		for _, config := range configs {
-			injectors = append(injectors, injector.NewCPUPressureInjector(spec, injector.CPUPressureInjectorConfig{Config: config}))
+			injectors = append(injectors, injector.NewCPUPressureInjector(spec, injector.CPUPressureInjectorConfig{Config: config, DistinctPodCores: cores}))
```

Comment:
I would no longer do a loop here, as you are already aggregating all configuration in your previous code, it seems to me that we'll do several time a pressure 🤔 

For every container, we'll launch as many goroutines as the total number of core used by all containers.

Not sure it's what we want 😬

Reply:
as per current logic, we are indeed doing several time a pressure, each pressure for a container.

> For every container, we'll launch as many goroutines as the total number of core used by all containers.
this is actually a valid point. In Jira, the ticket descriptions say we want to stress more efficiently by targeting all cores. We probably need to confirm if this is the expected behavior

Reply:
Did you had the opportunity to look at this:
* https://github.com/DataDog/chaos-controller/blob/main/docs/cpu_pressure.md

The expected behaviour, is, for each cpuset encountered among all containers of the Pod, we want a single stresser per cpuset.

Each stresser is taking a whole CPU down (100% cpu consumption) hence there is no need to have several stresser per CPU.

Reply:
I think you're correct @luphaz , thank you!